### PR TITLE
Use IndexOf{Any} in a few more places in System.Reflection.Metadata

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/MemoryBlock.cs [hunk 1]
```diff
@@ -315,33 +314,23 @@ internal string PeekUtf8NullTerminated(int offset, byte[]? prefix, MetadataStrin
...
             Debug.Assert(terminator <= 0x7f);
 
-            byte* start = Pointer + offset;
-            byte* end = Pointer + Length;
-            byte* current = start;
-
-            while (current < end)
+            ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(Pointer + offset, Length - offset);
+            int length = terminator != '\0' ?
+                span.IndexOfAny((byte)0, (byte)terminator) :
```
Comment (xtqqczze) [inline src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/MemoryBlock.cs:325]: Out-of-bounds read if `offset` is negative?

Comment (xtqqczze) [reply src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/MemoryBlock.cs:325]: Yep, `CheckBounds(offset, 0)` only validates against `Length`, and not `Pointer`.

Comment (stephentoub) [reply src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/MemoryBlock.cs:325]: Isn't that already the case?

Comment (xtqqczze) [reply src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/MemoryBlock.cs:325]: I see this is in fact validated in `CheckBounds` with some casting tricks:

```csharp
if (unchecked((ulong)(uint)offset + (uint)byteCount) > (ulong)Length)
```

**Pull request merged**